FeynHiggs-2.10.0/man/man1/FeynHiggs.1
.TH FEYNHIGGS 1 "1-Jun-2012"
.SH NAME
.PP
FeynHiggs \- the FeynHiggs command-line frontend
.SH SYNOPSIS
FeynHiggs parameterfile [flags]
.SH DESCRIPTION
FeynHiggs calculates the masses and couplings of the Higgs sector in the 
MSSM.  FeynHiggs provides a convenient way of invoking the subroutines 
of the FeynHiggs library from the command line.  The input parameters are 
read from a file and the output is displayed in a human-readable form on 
screen, but can be piped through a utility program to yield
machine-readable data files, e.g. for plotting.
.SH PARAMETER FILES
The input parameters for FeynHiggs are read from a file.  FeynHiggs
attempts to read an SLHA file first and if that fails, falls back onto 
its native format (description follows below).  Output-wise, the default 
is to write out an SLHA file (original filename + .fh-NNN) if the input 
was one, and write the output to screen otherwise.  Output style can be
changed by appending #SLHA to the filename to force SLHA output, and
#FH to force screen output.

Description of the native FeynHiggs format: comments begin with a #
and continue to the end of the line.  Apart from a possible comment,
each line has the format
.P
  var  lower  [upper]  [*][/][step]
.P
and declares a loop over
.I var
from
.I lower
to
.I upper
in steps of
.I step.
A "*" in front of 
.I step
results in a logarithmic stepping, i.e.
.I var
is multiplied by
.I step
in each iteration.  A prefix of "/" indicates that
.I step
is to be interpreted as the number of steps, rather than the step size.
.P
Simple assignments of the form
.I name lower
(i.e. no loop) are just a special case of this since the items in 
brackets may be omitted.
.P
.I var
is the name of the parameter, which is one of
.br
-------------------+--------------------------------------
.br
invAlfa            | the inverse of the electromagnetic
.br
                   | coupling constant
.br
-------------------+--------------------------------------
.br
AlfasMZ            | the strong coupling constant at MZ
.br
-------------------+--------------------------------------
.br
GF                 | the Fermi constant
.br
-------------------+--------------------------------------
.br
MS, MC, MT, MB     | the strange, charm, top, bottom masses
.br
-------------------+--------------------------------------
.br
MW, MZ             | the W and Z masses
.br
-------------------+--------------------------------------
.br
TB                 | tan(beta)
.br
-------------------+--------------------------------------
.br
MA0, MHp           | the CP-odd and charged Higgs mass
.br
                   | (only one should be specified)
.br
-------------------+--------------------------------------
.br
MSusy              | a generic soft-SUSY breaking mass
.br
                   | used as a default value for all
.br
                   | MgSL, MgSE, MgSQ, MgSU, MgSD.
.br
-------------------+--------------------------------------
.br
MgSL, MgSQ         | individual soft-SUSY breaking masses
.br
MgSE, MgSU, MgSD   | for the slepton and squark doublets
.br
g = 1..3           | and singlets.
.br
                   | Default: all MgSx = MSusy.
.br
-------------------+--------------------------------------
.br
Abs(MUE), Re(MUE)  | the Higgs mixing parameter
.br
Arg(MUE), Im(MUE)  |
.br
-------------------+--------------------------------------
.br
Abs(Af), Re(Af)    | the sfermion mixing parameters
.br
Arg(Af), Im(Af)    |
.br
f = e,mu,tau,      |
.br
    u,c,t,         |
.br
    d,s,b          |
.br
-------------------+--------------------------------------
.br
Abs(Xf), Re(Xf)    | alternative sfermion mixing parameter;
.br
Arg(Xf), Im(Xf)    | since Xf and Af are not independent,
.br
f = tau,t,b        | only one should be specified
.br
-------------------+--------------------------------------
.br
Abs(M_1), Re(M_1)  | the U1 gaugino mass parameter
.br
Arg(M_1), Im(M_1)  |
.br
-------------------+--------------------------------------
.br
Abs(M_2), Re(M_2)  | the SU2 gaugino mass parameter
.br
Arg(M_2), Im(M_2)  |
.br
-------------------+--------------------------------------
.br
Abs(M_3), Re(M_3)  | the gluino mass parameter
.br
Arg(M_3), Im(M_3)  |
.br
-------------------+--------------------------------------
.br
Qtau, Qt, Qb       | the scales at which the stau, stop,
.br
                   | and sbottom DRbar masses are defined
.br
                   | Default: all 0 (= on-shell masses)
.br
-------------------+--------------------------------------
.br
CKMlambda          | the CKM parameters
.br
CKMA               |
.br
CKMrhobar          |
.br
CKMetabar          | Default: all -1 (= use PDG values)
.br
-------------------+--------------------------------------
.br
Abs(deltaFXYij)    | the flavour-violating
.br
Arg(deltaFXYij)    | parameters
.br
Re(deltaFXYij)     |
.br
Im(deltaFXYij)     |
.br
F = L,E,Q,U,D      |
.br
XY = LL,LR,RL,RR   |
.br
ij = 12,23,13,     | Default: all 0
.br
-------------------+--------------------------------------
.br
scalefactor        | the ren. scale is MT*scalefactor
.br
-------------------+--------------------------------------
.br
prodSqrts          | sqrt(s) for the production x-sections
.br
-------------------+--------------------------------------
.P
Needless to add that complex quantities should be specified EITHER
through Abs and Arg OR through Re and Im.  In conflicting
cases a warning is issued and the Abs/Arg value takes
precedence.
.SH FLAGS
The flags are given as an optional 9-digit string on the command line,
where the default is 400202110.  The positions in the string are as
follows:
.TP
#1: mssmpart = 0..4
specifies the scope of the 1-loop part:
.br
0: MT^4-approximation
.br
1: top/stop-sector
.br
2: top/stop- + bottom/sbottom-sector
.br
3: full (s-)quark/lepton-sector
.br
4: full MSSM
.br
.B recommended: 4
.TP
#2: fieldren = 0..4
determines the 1-loop field-renormalization constants:
.br
0: DRbar
.br
1: On-Shell, Dabelstein's convention
.br
2: On-Shell, 'Goldstone-pole' version
.br
3: On-Shell, MF I
.br
4: On-Shell, MF II
.br
.B strongly recommended: 0
.TP
#3: tanbren = 0..2
determines the 1-loop tan(beta) counter term:
.br
0: DRbar
.br
1: field renormalization part only (Dabelstein)
.br
2: mixed field/On-Shell-renormalization
.br
.B strongly recommended: 0
.TP
#4: higgsmix = 1..3
determines the mixing in the Higgs sector:
.br
1: All non-diagonal (ND) Self Energies = 0
.br
2: All CP-violating ND Self Energies = 0
.br
3: Full 3x3 mixing in the neutral sector
.br
.B This means:
.br
.B 2: evaluation in the rMSSM
.br
.B 3: evaluation in the cMSSM
.TP
#5: p2approx = 0..4
determines the approximation for the 1-loop result:
.br
0: None, i.e. full determination of the propagator matrices's poles,
   UHiggs is evaluated at p^2 = m^2
.br
1: p^2 = 0 approximation
.br
2: All self energies are calculated at p^2 = m_tree^2
.br
3: imaginary parts of self-energies are discarded
.br
4: as 0, but with UHiggs evaluated at p^2 = 0
.br
.B recommended: 4
.TP
#6: looplevel = 0..2
determines the inclusion of higher-order corrections:
.br
0: tree level
.br
1: one-loop contributions only
.br
2: include various two-loop contributions
.br
.B recommended: 2
.TP
#7: runningMT = 0..1
determines which top mass shall be used in the 1-/2-loop corrections:
.br
0: pole MT
.br
1: running MT
.br
.B recommended: 1
.TP
#8: botResum = 0..1
determines whether the O(tan beta^n) corrections shall be resummed:
.br
0: no resummation
.br
1: resummation of MB
.br
.B recommended: 1
.TP
#9: tlCplxApprox = 0..6
determines how the two-loop corrections are treated in the presence of
complex parameters (cMSSM):
.br
0: all corrections (asat, asab, atat, atab) are computed in
   the rMSSM
.br
1: only the cMSSM asat corrections are used
.br
2: the cMSSM asat corrections are combined with the remaining
   corrections in the rMSSM
.br
3: the cMSSM a_s a_t corrections are combined with the remaining
   corrections, whose complex phases are interpolated in
   At, Ab, M_3, MUE
.br
4: ditto, with interpolation in At, Xb, M_3, MUE
.br
5: ditto, with interpolation in Xt, Ab, M_3, MUE
.br
6: ditto, with interpolation in Xt, Xb, M_3, MUE
.SH EXAMPLE
Using the example parameter file that comes with FeynHiggs, the
command
.P
  FeynHiggs example/var.in
.P
should result in the following Higgs masses and mixings:
.P
| Mh0           =     113.89008624
.br
| MHH           =     195.87658697
.br
| MA0           =     200.00000000
.br
| MHp           =     213.63955069
.br
| SAeff         =    -0.35592463    0.00000000
.br
| UHiggsRe      =     0.99672973    0.08080741    0.00000000
.br
|                    -0.08080741    0.99672973    0.00000000
.br
|                     0.00000000    0.00000000    1.00000000
.P
and the following widths and branching ratios:
.P
| h0-b-b        =    6.018044E-03   0.825117       0.725474    
.br
| HH-b-b        =    5.165486E-02   0.559950       2.526224E-03
.br
| A0-b-b        =    5.896187E-02   0.848461       2.280889E-03
.br
| Hp-t-b        =    6.224386E-02   0.781530    
.P
Before the results, FeynHiggs lists the parameters of the calculation.
.P
With all the detailed information on widths and branching ratios, the 
screen can become rather crowded.  To mask off this detailed output, 
type
.P
  FeynHiggs ... | grep -v %
.SH CONVERSION TO MACHINE-READABLE OUTPUT
The output of FeynHiggs can be converted into a format more amenable to 
further processing, e.g. plotting, by the "table" utility.  This works as 
follows:
.P
  FeynHiggs ... | table TB Mh0 > var.out
.P
var.out then contains two columns listing TB and Mh0 for each pass of 
the loop.
.SH SPS BENCHMARK SCENARIOS
The SPS benchmark scenarios are predefined.  The input files can be 
found in example/SPS/.  To run a scenario, type e.g.
.P
  FeynHiggs example/SPS/SPS1a.in
.P
For more information on the SPS, see
.P
  http://www.ippp.dur.ac.uk/~georg/sps/
.SH LES HOUCHES BENCHMARK SCENARIOS
The Les Houches benchmark scenarios for Higgs boson searches at hadron
colliders are predefined.  The input files can be found in 
example/LHBMS/.  To run a scenario, type e.g.
.P
  FeynHiggs example/SPS/mhmax.in
.P
For more information on the Les Houches Benchmark scenarios, see
Eur. Phys. J. C26 (2003) 601 [hep-ph/0202167].
.SH SUSY LES HOUCHES ACCORD DATA
FeynHiggs can handle data files in the SUSY Les Houches Accord (SLHA) 
format.  If the input file is in SLHA format, the results (masses,
widths, etc.) are equally written out in SLHA format, to a file named as 
the source file suffixed by ".fh".  For example:
.P
  FeynHiggs example/SLHA/TestEXTPAR.spc
.P
produces example/SLHA/TestEXTPAR.spc.fh.  FeynHiggs can process SLHA
files in which the input parameters are given (Block EXTPAR) as well as 
files in which the masses and mixings is given (Blocks MASS, STOPMIX, 
etc.), where in the latter case the input parameters are reconstructed
from the mass spectrum.  For instance, try:
.P
  FeynHiggs example/SLHA/SphenoRP.spc
.P
For more information on the SLHA, see
.P
  http://home.fnal.gov/~skands/slha/
.SH ENVIRONMENT
The FeynHiggs debug level can be set using the FHDEBUG environment 
variable, e.g.
.P
  setenv FHDEBUG 1   (in tcsh)
.br
  export FHDEBUG=1   (in bash)
.P
The pager spawned to view FeynHiggs output is taken from the environment 
variable PAGER and defaults to "less" if PAGER is not set.  Should a 
program abort occur with FeynHiggs, it is advisable to set PAGER to an 
empty string, e.g.
.P
  setenv PAGER
.P
This prevents a pager from being started and ensures that the output
is shown right up to the point where the abort occurs.
.SH SEE ALSO
.PP
libFH(1), MFeynHiggs(1)


pyslha-2.1.3/ChangeLog
2013-07-15  Andy Buckley  <andy.buckley@cern.ch>

	* Version 2.1.3

	* Allow a missing MASS block, if an extra ignorenomass kw argument
	is passed to the readSLHA(File) functions.

	* Fix **kwargs passing from file read/write functions to the
	string read/write ones.

	* Add parsing robustness against unknown data section types,
	particularly the proposed XSECTION.

2013-05-31  Andy Buckley  <andy.buckley@cern.ch>

	* Version 2.1.2

	* Adding a reference to the arXiv paper in the package docstring
	and slhaplot help message.

2013-05-10  Andy Buckley  <andy.buckley@cern.ch>

	* Version 2.1.1

	* Layout tweak for decay ID code alignment in SLHA output.

	* Treat consecutive strings as a single value in SLHA entry parsing.

	* Improvements/fixes to repr vs. str and tuple-value output formatting.

2013-05-07  Andy Buckley  <andy.buckley@cern.ch>

	* Version 2.1.0

	* Adding multi-value support and refactoring Block value setting.

2013-04-29  Andy Buckley  <andy.buckley@cern.ch>

	* Version 2.0.0 with new Block API.

	* Adding a leading space to float formatting of positive numbers
	to improve alignment.

	* Clean-up of precision formatting in output.

	* Split writeSLHA implementation into distinct writeSLHA{Blocks,Decays}.

2013-04-28  Andy Buckley  <andy.buckley@cern.ch>

	* Adding a dict arg (compatible with dict.update) to the Block
	constructor.

	* Adding Block.set_value(*args) and Block documentation.

	* Various fixes to ISAWIG output (and conversion to use new
	tuple-indexing on mixing matrices and single-value accessing on
	ALPHA).

	* Improved API for access to entries/values and single/multi-value
	state on Block.

2013-04-26  Andy Buckley  <andy.buckley@cern.ch>

	* Improved SLHA output formatting.

	* Adding direct entry access via the [] operator to the Block
	class's entries.

	* Converting entries storage to avoid recursion, by using tuples
	as dict keys if necessary.

	* Version 1.5.0, since the behaviour has significantly altered.

	* Preserving the ordering of blocks, decays, and their their
	contents if possible, using ordered dicts. Thanks to Anders
	Kvellestad for the suggestion.

2013-04-08  Andy Buckley  <andy.buckley@cern.ch>

	* Version 1.4.4

	* pyslha.py: Fix handling of particles with NaN widths (and decay
	channels with NaN BRs). Both are filled into the blocks in
	readSLHA(File) now, with None as the value for the invalid
	width/BR: test for correctness with e.g. 'myparticle.width is not
	None'. BR == NaN decays will be added if the ignorenobr arg ==
	False (the default). Thanks to Lukas Vanelderen for the report.

2012-11-16  Andy Buckley  <andy.buckley@cern.ch>

	* Version 1.4.2, 1.4.3

	* setup.py: Adding install_requires for tex2pix so that
	PyPI/easy_install/pip automatic dependency handling will work.

2012-03-19  Andy Buckley  <andy.buckley@cern.ch>

	* Adding a naive --aspect-ratio option, but the output is not ideal.

2012-03-04  Andy Buckley  <andy.buckley@cern.ch>

	* Version 1.4.1

	* Adding the --include command line option, which allows
	user-customisable additions to the plot.

	* Version 1.4.0

	* Updated test script which can take the output formats as an optional argument.

	* Check for duplicates and unsupported formats in the format string.

	* Adding rendering support for JPEG and PS (as opposed to EPS) formats.

	* Changing format string behaviour to require comma separators.

2012-03-03  Andy Buckley  <andy.buckley@cern.ch>

	* Version 1.3.3

	* Fix pyslha.ParseError printout to be more helpful.

	* Version 1.3.2

	* Improving error handling if tex2pix cannot be found or the MASS
	block is missing in the spectrum file, and adding the
	pyslha.ParseError class.

	* Version 1.3.1

	* Fix LaTeX package checks for tikz an mathpazo by adding .sty to
	the searched-for package names.

	* Version 1.3.0

	* Test for presence of the mathpazo package before including it in
	the default preamble.

	* Converting rendering system to use the separate tex2pix
	library (which itself is based on the slhaplot rendering
	experience).

2011-10-12  Andy Buckley  <andy@insectnation.org>

	* Version 1.2.9

	* Fixing decay style handling, thanks to a suggestion from John Conley.

2011-09-24  Andy Buckley  <andy@insectnation.org>

	* Version 1.2.8

	* Adding a max mass cutoff for slhaplot: don't show particle lines
	or associated decays for particles with masses greater than the
	cutoff. This is set to 10 TeV by default, i.e. above realistic LHC
	scales and was really introduced to protect the plotting code
	against Planck scale gravitons.

2011-09-10  Andy Buckley  <andy@insectnation.org>

	* Version 1.2.7 (I'm not sure what happened to the ChangeLog
	reports for 1.2.{4,5,6}...)

	* slhaplot: Fixing the min/max x/y functions in the class used to
	implement label merging. Thanks to Alexander Mann for the bug
	report, patch, and test spectrum!

2011-06-20  Andy Buckley  <andy@insectnation.org>

	* pyslha.py: Add "precision" arguments to the write* functions,
	with default FP precision set to ".8" as suggested by Diego
	Restrepo.

2011-04-10  Andy Buckley  <andy@insectnation.org>

	* slhaplot: Add a warning about lack of provided decays if decays
	should be drawn, but none have been found in the input file.

	* Version 1.2.3.

	* Fix insufficiently careful parsing of BLOCK "Q=" declarations.

	* Add little pre-release testing script, "test".

	* Add missing IDHW / IDPDG mapping for W+- and Z0.

2011-03-13  Andy Buckley  <andy@insectnation.org>

	* Version 1.2.2: use a more dynamic system for choosing y-tick
	values, to avoid overlaps and too-close tick labels in high-mass
	SUSY models.

2011-03-10  Andy Buckley  <andy@insectnation.org>

	* Version 1.2.1: add more semantic and case-insensitive handling
	of format codes, and simultaneously avoid using the any() function
	which requires Python >= 2.5.

2011-02-27  Andy Buckley  <andy@insectnation.org>

	* Version 1.2.0 (major version change due to interface changes and
	major new format functionality)


FeynHiggs-2.10.0/src/Frontend/FeynHiggs.F
* FeynHiggs.F
* the command-line front end for FeynHiggs
* this file is part of FeynHiggs
* last modified 18 Dec 13 th

#include "FHCouplings.h"
#include "types.h"
#include "debug.h"
#include "version.h"


	program FeynHiggs
	implicit none

#include "FHRecord.h"
#include "SLHA.h"

	character*128 argv
	integer argc, type, error, loop, keyslha, i
	integer mssmpart, fieldren, tanbren, higgsmix, p2approx
	integer looplevel, runningMT, botResum, tlCplxApprox
	RecordDecl(record)
	ComplexType slhadata(nslhadata)

#if U77EXT
#define len_trim lnblnk
	integer iargc, lnblnk
	external iargc, lnblnk
#endif

	character*16 flags

	character*55 dmssmpart(-1:5), dfieldren(-1:5),
     &    dtanbren(-1:3), dhiggsmix(0:5), dp2approx(-1:5),
     &    dlooplevel(-1:4), drunningMT(-1:2),
     &    dbotResum(-1:3), dtlCplxApprox(-1:7)

	data flags /"400242110"/

	data dmssmpart /
     &    "???",
     &    "MT^4 approximation (for real parameters only)",
     &    "top/stop sector",
     &    "top/stop + bottom/sbottom sector",
     &    "full (s)quark/(s)lepton sector",
     &    "full MSSM [recommended]",
     &    "???" /

	data dfieldren /
     &    "???",
     &    "DRbar field ren. [strongly recommended]",
     &    "on-shell field ren., Dabelstein convention",
     &    "on-shell field ren., `Goldstone-pole' version",
     &    "on-shell field ren., MF I",
     &    "on-shell field ren., MF II",
     &    "???" /

	data dtanbren /
     &    "???",
     &    "DRbar TB-ren. [strongly recommended]",
     &    "field renormalization part only for TB (Dabelstein)",
     &    "mixed field/on-shell renormalization for TB",
     &    "???" /

	data dhiggsmix /
     &    "???",
     &    "no mixing in the neutral Higgs sector",
     &    "2x2 (h0-HH) mixing in the neutral Higgs sector",
     &    "3x3 (h0-HH-A0) mixing in the neutral Higgs sector",
     &    "4x4 (G0-h0-HH-A0) mixing [experimental]",
     &    "???" /

	data dp2approx /
     &    "???",
     &    "no approximation [recommended]",
     &    "p^2 = 0 approximation",
     &    "self-energies are calculated at p^2 = M_tree^2",
     &    "imaginary parts of self-energies are discarded",
     &    "UHiggs is evaluated at p^2 = 0",
     &    "???" /

	data dlooplevel /
     &    "???",
     &    "tree level only",
     &    "include 1-loop corrections",
     &    "include 2-loop corrections",
     &    "include resummation of large logs",
     &    "???" /

	data drunningMT /
     &    "???",
     &    "use pole top mass",
     &    "use running top mass",
     &    "???" /

	data dbotResum /
     &    "???",
     &    "non-resummed MB in 1L/2L corrections",
     &    "resummed 1L-MB in 1L/2L corrections [recommended]",
     &    "resummed 2L-MB in 1L/2L corrections",
     &    "???" /

	data dtlCplxApprox /
     &    "???",
     &    "all corrections evaluated in the rMSSM",
     &    "a_s a_t in cMSSM only",
     &    "a_s a_t in cMSSM, rest in rMSSM",
     &    "a_s a_t in cMSSM, rest interpolated in At,Ab,M_3,MUE",
     &    "a_s a_t in cMSSM, rest interpolated in At,Xb,M_3,MUE",
     &    "a_s a_t in cMSSM, rest interpolated in Xt,Ab,M_3,MUE",
     &    "a_s a_t in cMSSM, rest interpolated in Xt,Xb,M_3,MUE",
     &    "???" /

	call pageron

	argc = iargc()
	if( argc .lt. 1 .or. argc .gt. 3 ) goto 999

	if( argc .ge. 2 ) call getarg(2, flags)
	call FHSetFlagsString(error, flags)

	call FHRetrieveFlags(error,
     &    mssmpart, fieldren, tanbren, higgsmix, p2approx,
     &    looplevel, runningMT, botResum, tlCplxApprox)

#define ShowFlag(t,f,lo,hi) t, f, d f(min(max(f,lo-1),hi+1))

101	format(A, " = ", I1, T20, "# ", A)
	print 101,
     &    ShowFlag("mssmpart",     mssmpart,     0, 4),
     &    ShowFlag("fieldren",     fieldren,     0, 4),
     &    ShowFlag("tanbren",      tanbren,      0, 2),
     &    ShowFlag("higgsmix",     higgsmix,     1, 4),
     &    ShowFlag("p2approx",     p2approx,     0, 4),
     &    ShowFlag("looplevel",    looplevel,    0, 3),
     &    ShowFlag("runningMT",    runningMT,    0, 1),
     &    ShowFlag("botResum",     botResum,     0, 2),
     &    ShowFlag("tlCplxApprox", tlCplxApprox, 0, 6)

	if( error .ne. 0 ) stop

	call getarg(1, argv)
	i = index(argv, "#")
	type = -1
	if( i .gt. 0 ) then
	  type = 2
	  if( argv(i+1:) .eq. "SLHA" ) type = 0
	  argv(i:) = " "
	endif

	call FHReadRecord(error, record, slhadata, argv)
* 0 = SLHA, 2 = FH file, all others are true errors
	if( ibclr(error, 1) .ne. 0 ) goto 1

#if SLHAPARA
	keyslha = -2
#else
	keyslha = -4
#endif
	if( type .eq. 0 .and. error .ne. 0 ) then
	  call SLHAClear(slhadata)
	  keyslha = -1
	endif

	if( type .eq. -1 ) type = error

	if( type .eq. 0 ) then
	  i = 0
	  if( argv .ne. "-" ) i = len_trim(argv) + 1
	endif

* turn on verbose mode:
	record(iAdmin,iStep) = 1

	loop = 0

2	call FHLoopRecord(error, record)
	if( error .eq. -2 ) goto 2
	if( error .ne. 0 ) goto 1

	loop = loop + 1

	call FHSetRecord(error, record)
	if( error .ne. 0 ) goto 2

	if( type .eq. 0 ) then
	  call FHOutputSLHA(error, slhadata, keyslha)
	  if( error .ne. 0 ) print *, "Error in FHOutputSLHA ", error
	  if( i .gt. 0 ) write(argv(i:),'(".fh-",I3.3)') loop
	  call SLHAWrite(error, slhadata, argv)
	else
	  call FHOutput(error, "stdout", -1, record(iprodSqrts,iVar))
	  if( error .ne. 0 ) print *, "Error in FHOutput ", error
	  print 200
	endif

	goto 2


#define DESC_FMT (/I5,": ",A)

200	format(/"-------------------- END OF OUTPUT ---------------------")

100	format ("FeynHiggs ", FHVERSION
     &    /"built on ", FHBUILDDATE
     &    //"USAGE:  ", A, " <inputfile> [<flags>]"
     &    //"<inputfile> is a file containing the parameters"
     &    //"<flags> is a string of 9 digits with"
     &    /"  #1 = mssmpart = scope of the calculation",
     &    5 DESC_FMT
     &    /"  #2 = fieldren = 1-loop field renormalization",
     &    5 DESC_FMT
     &    /"  #3 = tanbren = 1-loop tan(beta) renormalization",
     &    3 DESC_FMT
     &    /"  #4 = higgsmix = mixing in the neutral Higgs sector",
     &    4 DESC_FMT
     &    /"  #5 = p2approx = approximation for the 1-loop result",
     &    5 DESC_FMT
     &    /"  #6 = looplevel, inclusion of higher-order corrections",
     &    4 DESC_FMT
     &    /"  #7 = runningMT, pole vs. running MT",
     &    2 DESC_FMT
     &    /"  #8 = botResum, MB resummation",
     &    3 DESC_FMT
     &    /"  #9 = tlCplxApprox, complex parameters in the 2-loop corrections",
     &    7 DESC_FMT
     &    /)

999	call getarg(0, argv)
	print 100, Strip(argv),
     &    (i, dmssmpart(i),     i = 0, 4),
     &    (i, dfieldren(i),     i = 0, 4),
     &    (i, dtanbren(i),      i = 0, 2),
     &    (i, dhiggsmix(i),     i = 1, 4),
     &    (i, dp2approx(i),     i = 0, 4),
     &    (i, dlooplevel(i),    i = 0, 3),
     &    (i, drunningMT(i),    i = 0, 1),
     &    (i, dbotResum(i),     i = 0, 2),
     &    (i, dtlCplxApprox(i), i = 0, 6)

1	call flush(6)
	call pageroff
	end

